pkg/template/app/app.go

```diff
@@ -120,3 +130,64 @@ func NewSecret(config SecretConfig) (*corev1.Secret, error) {
 
 	return secret, nil
 }
+
+// printAppCR removes empty fields from the app CR YAML. This is needed because
+// although the fields are optional we do not use struct pointers. This will
+// be fixed in a future version of the App CRD.
```

Comment:
This is nasty but I don't see another way of stripping the empty fields. 

So I've tried to make sure there is good test coverage.

Reply:
Does the YAML issue still exist if using `sigs.k8s.io/yaml`?

https://github.com/giantswarm/fmt/blob/master/go/yaml_marshalling.md

Reply:
@axbarsan No sadly the empty fields are still shown :(

The problem is although we have JSON tags with`omitempty` we use regular structs not struct pointers.

Theo tried to introduce this in https://github.com/giantswarm/apiextensions/pull/283. It works well but the problem is we need to handle the nil structs and the code changes got big.

The safest way I can see to introduce these changes is with a new v1alpha2 version of the App CR.

Reply:
@axbarsan I've switched to use `sigs.k8s.io/yaml` to match fmt. We were mostly using that anyway.

I've created https://github.com/giantswarm/roadmap/issues/365 for making the fields optional in the App CRD by using struct pointers.

We already did this for the new Catalog CRD that replaces AppCatalog.